rtl/eclrtl/eclregex.cpp

```diff
@@ -389,16 +508,67 @@ class CCompiledStrRegExpr : implements ICompiledStrRegExpr
 // STRING implementation
 //---------------------------------------------------------------------------
 
+/**
+ * @brief Fetches or creates a compiled string regular expression object.
+ *
+ * This function fetches a compiled string regular expression object from the cache if it exists,
+ * or creates a new one if it doesn't. The regular expression object is created based on the provided
+ * regex pattern, length, and case sensitivity flag. The created object is then cached for future use.
+ *
+ * @param _regexLength      The length of the regex pattern.
+ * @param _regex            The regex pattern.
+ * @param _isCaseSensitive  Flag indicating whether the regex pattern is case sensitive or not.
+ * @return  A pointer to a copy of the fetched or created CCompiledStrRegExpr object.  The returned object
+ * *        must eventually be deleted.
+ */
+CCompiledStrRegExpr* fetchOrCreateCompiledStrRegExpr(int _regexLength, const char * _regex, bool _isCaseSensitive)
+{
+    if (compiledCacheEnabled)
+    {
+        CCompiledStrRegExpr * compiledObjPtr = nullptr;
+        uint32_t options = (_isCaseSensitive ? 0 : PCRE2_CASELESS);
+        hash64_t regexHash = HASH64_INIT;
```

Comment:
cleanup: 
`regexHash = RegexHashEntry::hashValue(_regexLength, _regex, options);
`
and elsewhere